pyproject.toml

```diff
@@ -27,6 +27,7 @@ dependencies = [
     "nibabel>=2.5.1",
     "nilearn>=0.7",
     "numpy>=1.16",
+    "robustica>=0.1.3",
```

Comment:
Adding this dependency makes a lot of sense, but I want to highlight to @tsalo & others that this will also install two additional dependencies: scikit-learn-extra & https://tqdm.github.io/
We've had issues with adding modestly supported modules in the past (😱duecredit 😱) so we'll want to keep an eye on all three of these, particularly in relation to #934.
 
Also `tqdm` is a progress bar module. If we're going to require it, we might want to think if there are others parts of the code where it might be useful.

Reply:
It looks like there's a problem with `scikit-learn-extra` in the Python 3.12 tests.

Reply:
I have an opinion here - I had a branch where I played with putting the progress bar in for `curvefit` (I think) cause it is slow and can be confusing when it just sits there - I think it worked nicely but got distracted. Once this is pulled in, I'll look into doing that again. I don't think it makes sense for loglin, but could anyway (being forward looking, maybe it will be slow someday on some data?)

Reply:
I like that idea. I've used tqdm before without issue (e.g., in NiMARE), so I'm all for adding progress bars. Not sure about applicability to loglin though. Don't we only loop over unique adaptive mask values there, rather than voxels?

Reply:
good point - yeah, makes no sense there. I hadn't even checked it. Curve fit can use it, maybe figure creation? Honestly, once it punches through the ICA steps its hustling so may not be needed elsewhere.